### ISSUE TITLE
MediaWiki stuck in a loop until OutOfMemory after upgrade from MW 1.41 to 1.42+

### ISSUE THREAD

**Issue by MedicineStorm** [multidimension-al/phpbbauth#15]
After upgrading from MediaWiki 1.41 to MediaWiki 1.42 (or 1.43), it appears to get stuck in a loop trying to authenticate me (or any user). The page stops trying to load after the server either runs out of memory or it exceeds the 60 execution time limit.

Site is using:
> PHP 8.3.6 (litespeed)
> ICU 69.1
> MySQL 5.7.23-23

error_log shows nothing but the OutOfMemory fatal error, or
`PHP Fatal error:  Maximum execution time of 60 seconds exceeded in ./includes/libs/rdbms/database/DatabaseMySQL.php on line 671`

I added a `error_log( $sql);` line in that file on that line just for debugging and tried again. error_log is full of:

`SELECT /* MediaWiki\User\User::load  */  actor_id,actor_user,actor_name  FROM mw2021_actor    WHERE actor_name = 'MedicineStorm'  LIMIT 1`  

`SELECT /* MediaWiki\User\User::loadFromDatabase  */  user_id,user_name,user_real_name,user_email,user_touched,user_token,user_email_authenticated,user_email_token,user_email_token_expires,user_registration,user_editcount,user_actor.actor_id  FROM mw2021_user JOIN mw2021_actor user_actor ON ((user_actor.actor_user = user_id))   WHERE user_id = 3  LIMIT 1`  

`SELECT /* MediaWiki\User\User::loadFromDatabase  */  user_id,user_name,user_real_name,user_email,user_touched,user_token,user_email_authenticated,user_email_token,user_email_token_expires,user_registration,user_editcount,user_actor.actor_id  FROM mw2021_user JOIN mw2021_actor user_actor ON ((user_actor.actor_user = user_id))   WHERE user_id = 3  LIMIT 1`

Just those 3 queries on repeat. 162110 error_log entries later, it times out. Disabling PhpbbAuth extension causes everything to work fine... Except, y'know, authentication. I have trouble logging in to MediaWiki because I typically rely on this extension, so I'll have to hack my own account to get a working MW login, haha!

I have a full site backup, so I can always restore to MediaWiki 1.41, but that's not a long-term solution unless I plan to stay at 1.41 forever and gain no new benefits of regular MW upgrades and security fixes.

**TL;DR: I love this extension. The features it has are perfect for my needs. Please help. I am willing to work with you on debugging, maintaining, donating funds, whatever.**

**Comment by MedicineStorm** [comment]
Disregard. Was able to trace this to an issue in Auth_remoteuser. See https://www.mediawiki.org/wiki/Extension_talk:Auth_remoteuser/Archive_2

Solution: in `./extensions/Auth_remoteuser/src/UserNameSessionProvider.php` near line 528, I changed 
```
if ( !$user->isAllowed( 'createaccount' ) ) {
	$disableSpecialPages += [ 'CreateAccount' => true ];
}
```
To
```
//if ( !$user->isAllowed( 'createaccount' ) ) {
//	$disableSpecialPages += [ 'CreateAccount' => true ];
//}
```
Because apparently this block causes an infinite loop in MW 1.42+ during authentication. Account creation is already disabled via group permissions. However, I have not yet tested if this solution interferes with the auto-creation of new accounts from Phpbb.

Regardless, I consider this issue closed, or rather an issue with Auth_remoteuser, not PhpbbAuth. Thanks again.
